ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -455,90 +441,92 @@ export class AccountCreation extends React.Component<AccountCreationProps, Accou
-                       onChange={value => {
-                         this.setState(fp.set(['address', 'state'], value));
-                       }}
+            <TextInput data-test-id='state' name='state' placeholder='State' value={state}
+                       onChange={value => this.updateAddress('state', value)}
                        style={{...styles.section, marginBottom: '0.5rem'}}/>
-            <TextInput id='streetAddress2' name='streetAddress2' placeholder='Street Address 2'
+            <TextInput data-test-id='streetAddress2' name='streetAddress2' placeholder='Street Address 2'
                        value={streetAddress2}
                        style={{...styles.section, marginRight: '2rem', marginBottom: '0.5rem'}}
-                       onChange={value => {
-                         this.setState(fp.set(['address', 'streetAddress2'], value));
-                       }}/>
-            <TextInput id='zip' name='zip' placeholder='Zip Code' value={zipcode}
-                       onChange={value => {
-                         this.setState(fp.set(['address', 'zipcode'], value));
-                       }}
+                       onChange={value => this.updateAddress('streetAddress2', value)}/>
+            <TextInput data-test-id='zip' name='zip' placeholder='Zip Code' value={zipCode}
+                       onChange={value => this.updateAddress('zipCode', value)}
                        style={{...styles.section, marginBottom: '0.5rem'}}/>
-            <TextInput id='city' name='city' placeholder='City' value={city}
-                       onChange={value => {
-                         this.setState(fp.set(['address', 'city'], value));
-                       }}
+            <TextInput data-test-id='city' name='city' placeholder='City' value={city}
+                       onChange={value => this.updateAddress('city', value)}
                        style={{...styles.section, marginRight: '2rem'}}/>
-            <TextInput id='country' placeholder='Country' value={country} style={styles.section}
-                       onChange={value => {
-                         this.setState(fp.set(['address', 'country'], value));
-                       }}/>
-          </div>
+            <TextInput data-test-id='country' placeholder='Country' value={country} style={styles.section}
+                       onChange={value => this.updateAddress('country', value)}/>
+          </FlexRowWrap>
         </Section>
         <Section header='Institutional Affiliation'>
           <label style={{color: colors.primary, fontSize: 16}}>
             Are you affiliated with an Academic Research Institution?
           </label>
           <div style={{paddingTop: '0.5rem'}}>
-            <RadioButton onChange={() => {this.setState({showInstitution: true}); }}
+            <RadioButton data-test-id='show-institution-yes'
+                         onChange={() => {this.updateInstitutionAffiliation('affiliation', '');
```

Comment:
Should probably clear both affiliation and institution for both Yes and No

Reply:
so since other option will be defaulted either by default or by user selecting option already do we need to clear both again

Reply:
I prefer to clear everything that we're not going to use out of the state, just in case